src/nvim/popupmnu.c

```diff
@@ -200,12 +201,10 @@ void pum_display(pumitem_T *array, int size, int selected, bool array_changed)
     return;
   }
 
-  // If there is a preview window at the top avoid drawing over it.
-  if (firstwin->w_p_pvw
-      && (pum_row < firstwin->w_height)
-      && (pum_height > firstwin->w_height + 4)) {
-    pum_row += firstwin->w_height;
-    pum_height -= firstwin->w_height;
+  // If there is a preview window at the above avoid drawing over it.
+  if (pum_row < above_row && pum_height > above_row) {
```

Comment:
the original patch checks for NULL:

    if (pvwin != NULL && pum_row < above_row && pum_height > above_row)